clients/cobol-lsp-vscode-extension/src/commands/CommentCommand.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+
+export function commentCommand() {
+    const activeEditor = vscode.window.activeTextEditor;
+    if (!activeEditor) {
+        return;
+    }
+    new ToggleComments(activeEditor).toggle();
+}
+
+export class ToggleComments {
+    constructor(private textEditor: vscode.TextEditor) { }
+
+    public toggle() {
+        const replacingList = [];
+        for (const selection of this.textEditor.selections)
+            replacingList.push(this.handleSelection(selection));
+        if (replacingList.length === 0) return;
+        this.textEditor.edit(editBuilder => {
+            for (const replacing of replacingList)
+                editBuilder.replace(replacing.selection, replacing.text)
+        });
+    }
+
+    private handleSelection(selection: vscode.Selection) {
+        const selectedLines = this.getSelectedLines(selection)
+        const textLines = selectedLines.map(it => it.text);
+        const anyNonComment = textLines
+            .map(getLineCommentStatus)
+            .some(it => it === LineCommentStatus.NON_COMMENT);
+        const action = anyNonComment ? commentLine : uncommentLine;
+        const processedLines = textLines.map(action);
+        const lineSeparator = this.textEditor.document.eol === vscode.EndOfLine.LF ? '\n' : '\r\n';
```

Comment:
So, it is a configuration of VSC, not of the file itself? Is it expected?

Reply:
That comes from VSC. Have no idea how it evaluates it. You can see it in the bottom right part:
![image](https://user-images.githubusercontent.com/60617592/151408654-9a07a6d7-2ce2-4380-8c28-17adfa7f7de9.png)
And even change manually.

Reply:
I just wasn't sure if we have any convention about this. I don't really care.

Reply:
Ok